proto/src/determined/trial/v1/trial.proto

```diff
@@ -25,6 +25,8 @@ message CheckpointWorkload {
   map<string, int64> resources = 5;
   // Total number of batches as of this workload's completion.
   int32 total_batches = 8;
+  // Id of the checkpoint in the database.
+  int32 id = 9;
```

Comment:
It turns out that warmStartCheckpointId is the db ID and not the UUID 😬 
so when we compare is this first checkpoint based on the warmStartCheckpointId as expected, we want it available

Reply:
yah that's unfortunate but can't we just make this a uuid and checkpoint warm start checkpoint id on the trials api to a uuid, since we just added it, instead of releasing an api just to break eventually? then `proto_get_trials_plus.sql` can just shim the id to a uuid pretty easily.

Reply:
ok I am going to remove this field, and replace `warmStartCheckpointId` with `warmStartCheckpointUuid` since this name can distinguish it from previous use in the API